src/js/components/Header/index.js

```diff
@@ -1,11 +1,51 @@
-import React from 'react';
-import ReactDOM from 'react-dom';
+// @flow
+import React, { Component } from 'react';
+import Logo from '../Logo';
+import Search from '../Search';
+import Poster from '../Poster';
+import SearchButton from '../SearchButton';
+import { views } from '../../helpers';
 
-const header =  React.createElement('div', {className: 'header'},
-  React.createElement('div', {className: 'title header__title'}, 'netflixroulette')
-);
+type Props = {
+  onSearch: (data: string, filter:string) => void,
+  toSearch: () => void,
+  view: string,
+  posterData: {
+    title: string,
+  },
+};
 
-ReactDOM.render(
-  header,
-  document.getElementById('header')
-);
+function topHeader(props: Props) {
+  const { view, toSearch } = props;
+  if ( view === views.POSTER ) {
```

Comment:
It seems like a good idea to have different content components, like `HeaderPoster` and `HeaderSearch`, to avoid making multiple identical checks in each sub-component - we can make it once on a top level. Plus- they render different content, so there is not much reused. What's your thoughts on that?